src/odemis/gui/model/__init__.py

```diff
@@ -1407,27 +1408,44 @@ def __init__(self, microscope):
         if model.MD_SAMPLE_CENTERS not in md:
             raise KeyError("Stage has no MD_SAMPLE_CENTERS metadata.")
         centers = md[model.MD_SAMPLE_CENTERS]
-        layout = sample_positions_to_layout(centers)
-        for row_idx, row in enumerate(layout):
-            for column_idx, name in enumerate(row):
-                scintillator_number = int(name.split()[-1])
-                layout[row_idx][column_idx] = scintillator_number
         # SAMPLE_SIZES contains the sizes of the scintillators
         if model.MD_SAMPLE_SIZES not in md:
             raise KeyError("Stage has no MD_SAMPLE_SIZES metadata.")
         sample_sizes = md[model.MD_SAMPLE_SIZES]
+        # Handle error cases for sample centers and sizes
+        if set(centers.keys()) != set(sample_sizes.keys()):
+            raise KeyError("MD_SAMPLE_CENTERS and MD_SAMPLE_SIZES metadata should have "
+                "the same keys.")
+        if not any(isinstance(i, float) for i in chain.from_iterable(centers.values())):
```

Comment:
That means "if there is not a single value which is float", then it's an error. I think you want to replace `not any` by `not all` (or `any(not isinstance...`

Reply:
Good catch! fixed